prog/analysis.go

```diff
@@ -76,7 +76,7 @@ func (s *state) analyzeImpl(c *Call, resources bool) {
 			}
 		case *BufferType:
 			a := arg.(*DataArg)
-			if a.Dir() != DirOut && len(a.Data()) != 0 {
+			if a.Dir() != DirOut && len(a.Data()) != 0 && !typ.IsCompressed() {
```

Comment:
This will cause expensive decompression for all images while we don't need it here.
Not sure if this happened in other places.

Overall this separation to Data/RawData, SetData/SetRawData seems to complicate things everywhere, while the only code that really cares about compression is image mutation procedure.

If we do:
```
func (t *BufferType) mutate(...) {
...
	case BufferCompressed:
		data := Decompress(a.Data())
		...
		a.SetData(Compress(data))
```

We won't need any of the Data/RawData/SetData/SetRawData changes and calling Data() will still be cheap as it is now, so no potential for such performance bugs. And one won't need to think if we need to use Data or RawData in each callsite in the prog package.